projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/kinit_authenticator.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+import tempfile
+from subprocess import call  # nosec
+
+from vdk.internal.core import errors
+from vdk.plugin.kerberos.base_authenticator import BaseAuthenticator
+
+log = logging.getLogger(__name__)
+
+
+class KinitGSSAPIAuthenticator(BaseAuthenticator):
+    """
+    A Kerberos authenticator that uses a 'kinit' call to obtain its ticket-granting ticket (TGT).
```

Comment:
Maybe mention that in order to use the class, you need to have kerberoes installed locally on the machine and kinit command working correctly ...